api/fetch_tags.py

```
@@ -0,0 +1,84 @@
+    
+    # 2. Setup the search query parameters for GitHub API
+    query = f'repo:{repo_name} label:"good first issue" state:open'
+    encoded_query = urllib.parse.quote(query)
+    url = f'https://api.github.com/search/issues?q={encoded_query}&per_page=30'
+    
+    # 3. Construct the HTTP request payload with a custom User-Agent header
+    req = urllib.request.Request(url, headers={'User-Agent': 'gsoc-org-finder'})
+    
+    try:
+        # Open an automated network stream using a secure Context Manager
+        with urllib.request.urlopen(req) as response:
```

Ruff (0.15.13)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)